assignments/array-methods.js

```diff
@@ -55,29 +55,86 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
+const fullName = [];
+
+runners.forEach(element => {
+    
+    fullName.push(`${element.first_name} ${element.last_name}`);
+});
+
+console.log("*** Q1 ***");
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+// Explicit approach
+const allCaps = runners.map(function(element) {
+    return element.first_name.toUpperCase();
+});
+
+// implicit approach
+// let allCaps = runners.map(element => element.first_name.toUpperCase());
+
+console.log("\n*** Q2 ***");
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+const largeShirts = runners.filter(function(element) {
+        return element.shirt_size === "L"
+});
```

Comment:
ES6 array function are cleaner, start to use them:
--> const largeShirts = runners.filter(runner => runner.shirt_size === "L");